mllib/src/main/scala/org/apache/spark/ml/util/Instrumentation.scala

```diff
@@ -161,8 +162,9 @@ private[spark] class Instrumentation private () extends Logging with MLEvents {
    * Logs an exception raised during a training session.
    */
   def logFailure(e: Throwable): Unit = {
-    val msg = e.getStackTrace.mkString("\n")
-    super.logError(msg)
+    val msg = new StringWriter()
+    e.printStackTrace(new PrintWriter(msg))
+    super.logError(msg.toString)
```

Comment:
I'm not clear what the change is... is it indentation? then I agree. Where else do we have the same issue?

Reply:
The original log loses the error message, in this example, it's the line 'java.lang.RuntimeException: Failed to finish the task'

And it loses suppressed and cause of the exception too, in this case, it's the lines 
'Caused by: java.io.FileNotFoundException: File is not found
at com.xxx.Test.test(Test.java:105)
...'